flake/dev/fmt.nix

```diff
@@ -66,6 +66,7 @@
           formatter.nixf-diagnose.options = [
             "--auto-fix"
             "--ignore=sema-unused-def-lambda-witharg-formal"
+            "--ignore=sema-primop-overridden"
```

Comment:
Did you find this being more annoyance than good? I ended up changing variable names to avoid the conflict in my own stuff.

Reply:
There is only one occurence that is annoying, it is the use of `path` here:
https://github.com/nix-community/nixvim/blob/2a3eece6afe262d690b608dc43547ec3a33abee9/ci/version-info/default.nix#L3

Renaming to something else is not that easy, because this file is used with `callPackage` I guess.

Reply:
We could change the call site to:

```nix
pkgs.callPackage ./version-info {
  pkgsPath = pkgs.path;
}
```

if we _really_ want to enforce this diagnostic.

Reply:
>  I ended up changing variable names to avoid the conflict in my own stuff.

For me these kinda lints are fine for variable bindings, but can be problematic with argument/parameter names. Especially if you need to maintain an API or interface with an external API.

If version-info's pkgs `path` is the only example where we have friction, I'm fine with a workaround (as above).

From a purely philosophical perspective, I'm always hesitant to _enforce_ lints that hint at potential problems but also flag valid code, unless you can explicitly suppress them inline.

Reply:
I have no strong opinion. I would be happy with all three of the followings:
- Changing the call site as @MattSturgeon suggested above
- Ignoring this specific file
- Ignoring this rule

@khaneliman do you have a preference?

Reply:
> For me these kinda lints are fine for variable bindings, but can be problematic with argument/parameter names. Especially if you need to maintain an API or interface with an external API.
> 
> If version-info's pkgs `path` is the only example where we have friction, I'm fine with a workaround (as above).
> 
> From a purely philosophical perspective, I'm always hesitant to _enforce_ lints that hint at potential problems but also flag valid code, unless you can explicitly suppress them inline.

As some of the first key users of these features, I would appreciate it if you could gather common production issues or suggestions for improvement. I sometimes overlook real-world scenarios while writing these linting rules, so your feedback is invaluable.